vector/v.info/print.c

```diff
@@ -197,31 +246,79 @@ void print_shell(struct Map_info *Map, const char *field_opt)
 
     map_type = Vect_maptype(Map);
 
-    fprintf(stdout, "name=%s\n", Vect_get_name(Map));
-    fprintf(stdout, "mapset=%s\n", Vect_get_mapset(Map));
-    fprintf(stdout, "location=%s\n", G_location());
-    fprintf(stdout, "project=%s\n", G_location());
-    fprintf(stdout, "database=%s\n", G_gisdbase());
-    fprintf(stdout, "title=%s\n", Vect_get_map_name(Map));
-    fprintf(stdout, "scale=1:%d\n", Vect_get_scale(Map));
-    fprintf(stdout, "creator=%s\n", Vect_get_person(Map));
-    fprintf(stdout, "organization=%s\n", Vect_get_organization(Map));
-    fprintf(stdout, "source_date=%s\n", Vect_get_map_date(Map));
+    switch (format) {
+    case PLAIN:
+        fprintf(stdout, "name=%s\n", Vect_get_name(Map));
+        fprintf(stdout, "mapset=%s\n", Vect_get_mapset(Map));
+        fprintf(stdout, "location=%s\n", G_location());
+        fprintf(stdout, "project=%s\n", G_location());
+        fprintf(stdout, "database=%s\n", G_gisdbase());
+        fprintf(stdout, "title=%s\n", Vect_get_map_name(Map));
+        fprintf(stdout, "scale=1:%d\n", Vect_get_scale(Map));
+        fprintf(stdout, "creator=%s\n", Vect_get_person(Map));
+        fprintf(stdout, "organization=%s\n", Vect_get_organization(Map));
+        fprintf(stdout, "source_date=%s\n", Vect_get_map_date(Map));
+        break;
+    case JSON:
+        json_object_set_string(root_object, "name", Vect_get_name(Map));
+        json_object_set_string(root_object, "mapset", Vect_get_mapset(Map));
+        json_object_set_string(root_object, "location", G_location());
+        json_object_set_string(root_object, "project", G_location());
+        json_object_set_string(root_object, "database", G_gisdbase());
+        json_object_set_string(root_object, "title", Vect_get_map_name(Map));
+        json_object_set_number(root_object, "scale",
+                               Vect_get_scale(Map)); // fixme: 1:scale
```

Comment:
This is not used much lately. Does anyone want to dive into old doc to see what would be be proper representation of this in JSON? Library stores N (int) which apparently should be interpreted as 1:N scale. I guess I would vote for a string `1:N` in JSON, however we could also go with what the library does and that's storing the N as a number and relying on display to represent that properly to the users.

Reply:
I also vote for the `1:N` string format.

Reply:
Updated